juju/juju
# WIP: charm lxd profile used by containers with non lxd provider

## Description of change

When deploying an lxd container, write out any lxd profiles defined in the charm(s) for that container and specify them for use at container creation.

Tests to be written.

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=lxd-profile
2. juju bootstrap aws
3. juju deploy ./testcharms/charm-repo/quantal/lxd-profile --to lxd
4. wait for deploy to finish, get container name
4. juju ssh 0; sudo adduser ubuntu lxd; newgrp lxd
4. lxc profile show juju-default-lxd-profile-0
profile should exist and be a superset (without comments) as what's in ./testcharms/charm-repo/quantal/lxd-profile/lxd-profile.yaml, adding name and used_by.  Container name from step above should be listed in used_by.
5. unit should deploy per normal, charm profile is used.
## Documentation changes

*Please replace with any notes about how it affects current user workflow? CLI? API?* 

## Bug reference

*Please add a link to any bugs that this change is related to.*
